test/integration_test.py

```diff
@@ -714,3 +724,60 @@ def test_mini_dss_failures(self):
                 mini_dss.get_file(uuid, version, 'aws')
             with self.assertRaises(self.SpecialError):
                 mini_dss.get_native_file_url(uuid, version, 'aws')
+
+
+class AzulChaliceLocalIntegrationTest(AzulTestCase):
+    service_app = Path(config.project_root) / 'lambdas' / 'service'
+    host = '0.0.0.0'
+    port = 8000
+    local_chalice_server = None
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        localfactory = factory.CLIFactory(str(cls.service_app))
+        config = localfactory.create_config_obj()
+        cls.local_chalice_server = localfactory.create_local_server(app_obj=app,
+                                                                    config=config,
+                                                                    host=cls.host,
+                                                                    port=cls.port)
+        thread = threading.Thread(target=cls.local_chalice_server.serve_forever)
+        thread.start()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.local_chalice_server.shutdown()
+        super().tearDownClass()
+
+    def test_local_chalice_health_endpoint(self):
+        base_url = furl(scheme='http',
+                        host=self.host,
+                        port=self.port,
+                        path='health')
+        response = requests.get(base_url.url)
+        self.assertEqual(response.status_code, 200)
+
+    catalogs = config.catalogs.keys() - config.integration_test_catalogs.keys()
+
+    def test_local_chalice_index_endpoints(self):
+        for catalog in self.catalogs:
+            with self.subTest(catalog=catalog):
+                base_url = furl(scheme='http',
```

Comment:
Split this up using

```
url = furl(host='foo', port=123)
url.set(path='bar', query=dict(x='y'))
furl('//foo:123/bar?x=y')
```

and extract the common part into a class attribute.

Reply:
Also why is the variable called `base_url` and not just `url`? "Base" suggests that something is appended to it which, in your current version, is not the case.

Reply:
I misconceived the convention for naming urls.